rs/limits/src/lib.rs

```diff
@@ -41,3 +45,28 @@ pub const SMALL_APP_SUBNET_MAX_SIZE: usize = 13;
 
 /// Cycles threshold to reduce logging load for canister operations with cycles.
 pub const LOG_CANISTER_OPERATION_CYCLES_THRESHOLD: u128 = 100_000_000_000;
+
+pub const KILOBYTE: u64 = 1024;
+pub const MEGABYTE: u64 = KILOBYTE * KILOBYTE;
+
+pub const UNIT_DELAY_APP_SUBNET: Duration = Duration::from_millis(1000);
+pub const UNIT_DELAY_NNS_SUBNET: Duration = Duration::from_millis(3000);
+pub const INITIAL_NOTARY_DELAY_APP_SUBNET: Duration = Duration::from_millis(600);
+pub const INITIAL_NOTARY_DELAY_NNS_SUBNET: Duration = Duration::from_millis(1000);
+pub const MAX_INGRESS_MESSAGES_PER_BLOCK: u64 = 1000;
+pub const MAX_BLOCK_PAYLOAD_SIZE: u64 = 4 * MEGABYTE;
+/// This sets the upper bound on how big a single ingress message can be, as
+/// allowing messages larger than around 3.5MB has various security and
+/// performance impacts on the network.  More specifically, large messages can
+/// allow dishonest block makers to always manage to get their blocks notarized;
+/// and when the consensus protocol is configured for smaller messages, a large
+/// message in the network can cause the finalization rate to drop.
+pub const MAX_INGRESS_BYTES_PER_MESSAGE_APP_SUBNET: u64 = 2 * MEGABYTE;
```

Comment:
So far, this set of constants/limits has been used by ic-prep, i.e., in testnets etc.

Moving these constants here into a crate called `ic-limits` somewhat suggests that these _are_ the actual limits, e.g., the ones used in production, while in fact the actual limits of the IC used in production are (at least partially) defined elsewhere and the constants/limits defined here are rather default-values, e.g., used for testing (and possibly if a new subnet is created, but not sure).

For example, max ingress bytes per message in an App-subnet doesn't come from here from `MAX_INGRESS_BYTES_PER_MESSAGE_APP_SUBNET` but from the `max_ingress_bytes_per_message` field in the subnet record of the respective subnet in the registry.

Maybe all that's needed is to extend the Rustdoc a bit to explain what I said above. WDYT?

Reply:
Did you see the module comment at the top of the file ?

The observation is correct, however, ideally what we want is that the default values are defined here and ideally those are the values what we use in production. We should use the registry as an override. 

One of the problems that I am trying to solve is that people assume that a registry value always exists so there is either a panic or a fallback to some unvetted value. Ideally, we use sane defaults always and we use the registry for override in very specific situations.

@fspreiss  Please let me know if I should extend the module comment

Reply:
> Did you see the module comment at the top of the file ?

Yes, I saw it (before writing my comment), and the comment somewhat suggests that these are the limits used in production as of today ("The file contains system limits of the IC").

> The observation is correct, however, ideally what we want is that the default values are defined here and ideally those are the values what we use in production. We should use the registry as an override.

Ah, I see. So what is described in the comment is more the place where we want to be in the future, but not  necessarily where we are now. Potentially the comment could be adjusted to make this a bit more clear.

Also, maybe the part "this can happen via a registry proposal" can be made more specific by explaining the  idea/mechanism with the defaults and the registry overrides (i.e., that the idea is that the values here are defaults and that they are possibly overridden with a respective entry in the registry). Because someone who doesn't know the system so well could think that with a registry proposal the Rust constants here can be changed.